src/announcement/announcement.ts

```diff
@@ -0,0 +1,81 @@
+//eslint-disable-next-line
+require("dotenv").config();
+import Web3 from "web3";
+import { Contract } from "web3-eth-contract";
+import { AbiItem } from "web3-utils";
+
+import { KeccakHash } from "../types/hash";
+import { hashPrefix } from "../utilities/hashPrefix";
+
+const BATCH_CONTRACT_ADDRESS = process.env.BATCH_CONTRACT_ADDRESS as string;
+
+export interface BatchArguments {
+  provider: Web3;
+  accountAddress: string;
+  uri: string;
+  hash: KeccakHash;
+}
+const BATCH_CONTRACT_ABI: AbiItem[] = [
+  {
+    anonymous: false,
+    inputs: [
+      {
+        indexed: false,
+        internalType: "bytes32",
+        name: "hash",
+        type: "bytes32",
+      },
+      {
+        indexed: false,
+        internalType: "string",
+        name: "dsnpUri",
+        type: "string",
+      },
+    ],
+    name: "DSNPBatch",
+    type: "event",
+  },
+  {
+    inputs: [
+      {
+        internalType: "bytes32",
+        name: "hash",
+        type: "bytes32",
+      },
+      {
+        internalType: "string",
+        name: "dsnpUri",
+        type: "string",
+      },
+    ],
+    name: "batch",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+];
+
+const contract: Contract | null = null;
+
+export const getContract = (web3Instance: Web3): Contract => {
+  if (contract) return contract;
+  return new web3Instance.eth.Contract(BATCH_CONTRACT_ABI, BATCH_CONTRACT_ADDRESS);
```

Comment:
`BATCH_CONTRACT_ADDRESS` should likely be able to be passed in but default back to something. Any contract that matches the `IAnnounce` interface can be used.

Reply:
What would it default to? There isn't any one batch contract address that will work across everyone's docker nodes.

Reply:
I'm thinking of @aneyzberg's work with testnet contract discovery. If the sdk knows what network the user is on, then it should also know what the default contract address is.

The sdk can know which network as the sdk will need to know what chain id. Chain Id will map to the network. (testnet vs betanet vs mainnet)

Reply:
@wilwade however doesn't this depend on the local network being deployed the exact same way every time? So that the address of the contract does not change

Reply:
Not once we have our discovery code working.

Reply:
oh yeah. I suppose this is true.

Reply:
but I would say let's set a default once we actually have this discovery code working? Since currently setting a default would break things.